pthread/pthread.c

```diff
@@ -483,44 +583,138 @@ int pthread_setschedparam(pthread_t thread, int policy,
 int pthread_mutex_init(pthread_mutex_t *mutex, const pthread_mutexattr_t *attr)
 {
 	(void)attr;
-
-	return mutexCreate(mutex);
+	mutex->initialized = 1;
+	return mutexCreate(&mutex->mutexh);
 }
 
 
 int pthread_mutex_lock(pthread_mutex_t *mutex)
 {
-	if (mutex == NULL)
-		return -EINVAL;
+	int err = EOK;
 
-	return mutexLock(*mutex);
+	if (mutex == NULL) {
+		err = -EINVAL;
+	}
+	else {
+		if (mutex->initialized == 0) {
+			mutexLock(pthread_common.mutex_cond_init_lock);
+			if (mutex->initialized == 0) {
+				mutex->initialized = 1;
+				err = mutexCreate(&mutex->mutexh);
```

Comment:
set initialized to 1 only if err == EOK. We might've got -EINTR or some temporary error here, so it might succeed on the next try. Applies to other lazy constructors too